src/web/index.ts

```diff
@@ -24,20 +24,21 @@ export class Reader {
   }
 
   // read length (bytes) starting from offset (bytes)
-  // callback(err, buffer)
-  read(offset: number, length: number, cb: Callback<Buffer>): void {
-    const reader = new FileReader();
-    reader.onload = function () {
-      reader.onload = null;
-      reader.onerror = null;
-      cb(null, Buffer.from(reader.result as ArrayBuffer));
-    };
-    reader.onerror = function () {
-      reader.onload = null;
-      reader.onerror = null;
-      cb(reader.error ?? new Error("Unknown FileReader error"));
-    };
-    reader.readAsArrayBuffer(this._blob.slice(offset, offset + length));
+  async read(offset: number, length: number): Promise<Buffer> {
+    return await new Promise((resolve, reject) => {
```

Comment:
There's a newer Blob method `.arrayBuffer()` that returns a promise. However, jsdom doesn't support it so it breaks our unit tests 🤦 https://github.com/jsdom/jsdom/issues/2555  Also only somewhat newer browsers support it – so perhaps this is a sign the world is not yet ready for cleaner code.